src/core/exception_hacks.cc

```diff
@@ -71,6 +71,8 @@ module seastar;
 #endif
 
 namespace seastar {
+
+#ifndef SEASTAR_NO_EXCEPTION_HACK
```

Comment:
> restore SEASTAR_NO_EXCEPTION_HACK behavior

i'd suggest extract this into its own commit. and explain why we want to bring it back. and how this change helps.